src/hotspot/share/gc/shenandoah/shenandoahFullGC.cpp

```diff
@@ -380,26 +381,8 @@ class ShenandoahPrepareForCompactionTask : public AbstractGangTask {
   ShenandoahHeap*           const _heap;
   ShenandoahHeapRegionSet** const _worker_slices;
 
-public:
-  ShenandoahPrepareForCompactionTask(PreservedMarksSet *preserved_marks, ShenandoahHeapRegionSet **worker_slices) :
-    AbstractGangTask("Shenandoah Prepare For Compaction"),
-    _preserved_marks(preserved_marks),
-    _heap(ShenandoahHeap::heap()), _worker_slices(worker_slices) {
-  }
-
-  static bool is_candidate_region(ShenandoahHeapRegion* r) {
-    // Empty region: get it into the slice to defragment the slice itself.
-    // We could have skipped this without violating correctness, but we really
-    // want to compact all live regions to the start of the heap, which sometimes
-    // means moving them into the fully empty regions.
-    if (r->is_empty()) return true;
-
-    // Can move the region, and this is not the humongous region. Humongous
-    // moves are special cased here, because their moves are handled separately.
-    return r->is_stw_move_allowed() && !r->is_humongous();
-  }
-
-  void work(uint worker_id) {
+  template<bool ALT_FWD>
```

Comment:
Again, keep this in place, do `public`/`private` declarations here. Would help Shenandoah backports.